### DISCUSSION TITLE
Debugging Stylus online

### DISCUSSION THREAD

**Discussion by stdedos** [stylus/stylus#2829] [Q&A] [answered]
I am trying to see why https://github.com/openstyles/stylus/ raises this particular error:

![image](https://github.com/stylus/stylus/assets/133706/5cfbeba0-bcfb-471c-835d-6cb96810f2d2)

(PS: _Now I know why_ - but the error is super un-intuitive)

I am trying to use https://stylus-lang.com/try.html - but the result I get is even more baffling:

![image](https://github.com/stylus/stylus/assets/133706/f48635b5-5182-4fec-bca0-0437db18562e)

_playground proceeds to re-write my whole CSS-snippet_ 😕 

Using https://codebeautify.org/stylus-compiler, instead "just works" (I don't get any error)

Could https://stylus-lang.com/try.html improve by presenting errors in a more consolidated and understandable manner?

**Comment by vendethiel** [comment]
Those are two different Stylus

**Reply by stdedos** [reply]
So ... the https://github.com/openstyles/stylus/, when processes files with `@preprocessor   stylus`,  ... it uses a different `stylus` than this (https://github.com/stylus/stylus)? 😕

**Comment by slightlyfaulty** [comment]
I don't think there's any error with your stylus code. You should ask on https://github.com/openstyles/stylus/ for help on your issue.

**Reply by stdedos** [reply]
https://stylus-lang.com/docs/variables.html shows `font-size = 14px`
Is `font-size = 14px;` also valid?

_I'm not sure how/where to dig if Stylus has preferences (or mandates) for `;`s_

**Reply by slightlyfaulty** [reply, marked as answer]
Semi-colons are optional. The stylus playground isn't giving you an error, that's just how it shows your compiled CSS output.

https://stylus-lang.com/docs/#what-if-it-were-all-optional

**Reply by stdedos** [reply]
My bad. `red` looked like error 😅

**Reply by stdedos** [reply]
Oh. _Again it seems to be https://github.com/openstyles/stylus/issues/880_ 🤦
